custom_libs/subliminal_patch/providers/jimaku.py

```diff
@@ -0,0 +1,220 @@
+                derived_anilist_id = obj[0]["anilist_id"]
+            except Exception as e:
+                logger.error(f"Could not derive anilist_id: {str(e)}")
+                
+        url_search_param = None
+        if derived_anilist_id:
+            logger.info(f"Will search for entry based on anilist_id: {derived_anilist_id}")
+            url_search_param = f"anilist_id={derived_anilist_id}"
+        else:
+            logger.info(f"Will search for entry based on series_name: {series_name}")
+            url_search_param = f"query={urllib.parse.quote_plus(series_name)}"
+        
+        # Search for entry
+        url = f"{self.api_url}/entries/search?{url_search_param}"
+        response = self.session.get(url, timeout=10)
+        response.raise_for_status()
+        
+        data = response.json()
+        logger.debug(f"Length of response on {url}: {len(data)}")
+        if len(data) == 0:
+            logger.error(f"No entries have been returned.")
+            return None
+        
+        # Determine entry
+        entry_id = None
+        anilist_id = None
+        name_is_japanese = self._is_string_japanese(f"series_name")
+        
+        for entry in data:
+            if not derived_anilist_id:
+                entry_has_jp_name_field = 'japanese_name' in entry
+                dict_field = 'japanese_name' if (name_is_japanese and entry_has_jp_name_field) else 'english_name'
+                logger.debug(f"Attempting to get entry based on '{dict_field}' because name_is_japanese: {name_is_japanese} and entry_has_jp_name_field: {entry_has_jp_name_field}")
+            else:
+                dict_field = 'anilist_id'
+            
+            # Only match first entry
+            entry_property_value = entry.get(dict_field, None)
+            if entry_property_value is not None:
+                entry_id = entry.get('id')
+                anilist_id = entry.get('anilist_id', None)
+                logger.info(f"Matched entry: ID: '{entry_id}', anilist_id: '{anilist_id}', name: '{entry.get('name')}', english_name: '{entry.get('english_name')}'")
+                break
+            
+        if entry_property_value is None:
+            logger.error('Matched no entries.')
+            return None
+        
+        # Get a list of subtitles for entry
+        url = f"{self.api_url}/entries/{entry_id}/files?episode={episode_number}"
+        response = self.session.get(url, timeout=10)
+        response.raise_for_status()
+
+        data = response.json()
+        logger.debug(f"Length of response on {url}: {len(data)}")
+        if len(data) == 0:
+            logger.error(f"No subtitles have been returned by entry '{entry_id}' for episode number {episode_number}.")
+            return None
+        
+        # Determine which subtitle to use
+        # Unfortunatley, subtitles will have differing name schemes
+        # For that reason, we'll just try our luck with fuzyy matching
```

Comment:
I think should return an array with all subtitles found and let the scoring judge which one is better.